swupd/manifest.go

```diff
@@ -658,6 +671,10 @@ func (m *Manifest) subtractManifests(m2 *Manifest) {
 	for _, mi := range m2.Header.Includes {
 		m.subtractManifests(mi)
 	}
+
+	for _, mi := range m2.Header.Optional {
+		m.subtractManifests(mi)
```

Comment:
so the process of subtracting is only to remove files from a manifest  if they are contained in an included manifest to reduce the size of a manifest file?

Reply:
General update speed and size improvements since it also impacts pack creation and pack size. But yes in this case you can not do the subtraction because they are optional and content in the bundle must exist in order to do things like satisfy runtime dependencies.

Reply:
ok, thanks for the explanation